### DISCUSSION TITLE
Documentation nitpicking

### DISCUSSION THREAD

**Discussion by mviereck** [ImageMagick/ImageMagick#5844] [General]
Occasionally I stumble over minor documentation issues. Mostly not worth to open a ticket only for this.
However, I've decided to start this discussion to collect them, so they can be fixed/improved if desirable.
I'll write an "answer" for each one in case single points cause more discussion than expected at first.

**Comment by mviereck** [comment]
_EDIT: done_

The documentation of `-blur` at https://imagemagick.org/script/command-line-options.php#blur says: 

> The Radius is only used to determine the size of the array which will hold the calculated Gaussian distribution. It should be an integer. If not given, or set to zero, IM will calculate the largest possible radius that will provide meaningful results for the Gaussian distribution. 

Similar for `-unsharp`:
> For reasonable results, radius should be larger than sigma. Use a radius of 0 to have the method select a suitable radius.

This is likely true for some related options, too, and should be mentioned there, too. 

Already documented for:
 - `-gaussian-blur`
 - `-unsharp`
 
Missing at:
 - `-adaptive-blur`: https://imagemagick.org/script/command-line-options.php#adaptive-blur
 - `adaptive-sharpen`: https://imagemagick.org/script/command-line-options.php#adaptive-sharpen
 - `-selective-blur`: https://imagemagick.org/script/command-line-options.php#selective-blur
 - `-sharpen`: https://imagemagick.org/script/command-line-options.php#sharpen
 - others?

Note that I did not check them all yet. Yet checked: `-adaptive-blur` seems to accept radius `0`.

**Reply by mviereck** [reply]
Yet found another one for radius `0` documentation:
`-canny` https://imagemagick.org/script/command-line-options.php#canny

**Comment by mviereck** [comment]
_EDIT: done_

Double use of word `provides` at https://www.imagemagick.org/Usage/convolve/#unsharpen
> The full "-unsharp" operator provides provides another type of control.

**Comment by mviereck** [comment]
_EDIT: done_

`-wavelet-denoise` accepts an argument `softness`. https://imagemagick.org/script/command-line-options.php#wavelet-denoise

> Softness attenuates the threshold and typically ranges from 0.0 (none) to 1.0. The higher the value the more noise that remains in the image.

No default value for `softness` is given that would give the same result as if not specified.
I was not able to produce any useful result if setting `softness` to some values, so a working default would help to have a starting point.
It might as well be a bug that I don't get useful results if setting `softness`.

**Reply by urban-warrior** [reply]
Check the latest documentation for your suggestions.

**Reply by mviereck** [reply]
Thank you!
All is done, and `-wavelet-denoise` gives the same result with now documented value `0` for softness.

My once seen issues with unusable results for all values for softness are gone, so that was likely a bug that is fixed meanwhile.

It might be added that `softness` also takes percent values.

**Comment by mviereck** [comment]
In the documentation of `-fx`: https://imagemagick.org/script/fx.php

```
 gauss(x)
    gaussian function; gauss(x)=exp(-x*x/2)/sqrt(2*pi)
```
Maybe I misunderstand something because I am not familiar with `-fx`, but this looks like `gauss()` expects 1 argument.
But it seems to expect two arguments:
```
$ magick -size 400x400 radial-gradient: -fx "gauss(u)"  show:
magick: For function 'gauss' expected 2 arguments, found too few (1) at '' @ error/fx.c/GetFunction/1896.
```

**Reply by urban-warrior** [reply]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT main branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://imagemagick.org/archive/beta/ by sometime tomorrow.

**Comment by snibgo** [comment]
This is a bug in fx.c. The line that says:
```
  {fGauss,   "gauss" , 2},
```
... should be:
```
  {fGauss,   "gauss" , 1},
```

**Comment by mviereck** [comment]
`-filter`: https://imagemagick.org/script/command-line-options.php#filter
It seems that a `-filter` setting can be set back to default with `+filter`. But this is not declared in the documentation.

Edit:
This seems to be true as well for `-interpolate`. https://imagemagick.org/script/command-line-options.php#interpolate
Furthermore, the documentation does not tell the default value of  `-interpolate`.
